dev/tools/dartdoc.dart

```diff
@@ -300,30 +302,46 @@ void cleanOutSnippets() {
   }
 }
 
-File precompileSnippetsTool() {
+Future<File> precompileSnippetsTool(
+  String pubExecutable,
+  Map<String, String> pubEnvironment,
+) async {
   final File snapshotPath = File(path.join('bin', 'cache', 'snippets.snapshot'));
   print('Precompiling snippets tool into ${snapshotPath.absolute.path}');
   if (snapshotPath.existsSync()) {
     snapshotPath.deleteSync();
   }
+
+  Process process = await Process.start(
+    pubExecutable,
+    <String>['get'],
+    workingDirectory: kSnippetsRoot,
+    environment: pubEnvironment,
+  );
+  printStream(process.stdout, prefix: 'pub:stdout: ');
+  printStream(process.stderr, prefix: 'pub:stderr: ');
+  final int code = await process.exitCode;
+  if (code != 0)
+    exit(code);
+
   // In order to be able to optimize properly, we need to provide a training set
   // of arguments, and an input file to process.
   final Directory tempDir = Directory.systemTemp.createTempSync('dartdoc_snippet_');
   final File trainingFile = File(path.join(tempDir.path, 'snippet_training'));
   trainingFile.writeAsStringSync('```dart\nvoid foo(){}\n```');
-  Process.runSync(Platform.resolvedExecutable, <String>[
+  ProcessResult result = Process.runSync(Platform.resolvedExecutable, <String>[
     '--snapshot=${snapshotPath.absolute.path}',
     '--snapshot_kind=app-jit',
     path.join(
-      'dev',
-      'snippets',
       'lib',
       'main.dart',
     ),
     '--type=sample',
     '--input=${trainingFile.absolute.path}',
     '--output=${path.join(tempDir.absolute.path, 'training_output.txt')}',
-  ]);
+  ], workingDirectory: path.absolute(kSnippetsRoot));
+  if (result.exitCode != 0)
+    throw Exception('Could not generate snippets snapshot:\n${result.stderr}');
```

Comment:
You also want to print the stderr and stdout to their respective file descriptors:
```dart
stdout.write(result.stdout);
stderr.write(result.stderr);
```